sympy/core/sympify.py

```diff
@@ -91,7 +91,7 @@ def _convert_numpy_types(a, **sympify_args):
 
 
 def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
-        evaluate=None) -> Basic:
+        evaluate=None) -> Union['Basic', Any]:
```

Comment:
We could give a better type for the `_sympify` function.

Another issue is that we need a type that corresponds to "sympifiable" which is somewhat analogous to numpy's "array-like". So then for `_sympify` we have something like:
```python
def _sympify(arg: Sympifiable) -> Basic
```

Reply:
Yeah that's definitely a thing that should be investigated, and which is [very doable](https://docs.python.org/3/library/typing.html#user-defined-generic-types). I'm not sure I'd be up to the task of it though... my main aim is improving performance of the inline documentation in VS Code, so although I do want to be as accurate as I can be, I can't say that I'd feel capable of defining my own type for a `Sympifiable` since I'm still not as familiar with SymPy as I'd like to be.

I feel like adding more complex type hints like that is something to look for in the long-term, but I don't think I'll be up to the task until I've worked with the library for a bit longer.